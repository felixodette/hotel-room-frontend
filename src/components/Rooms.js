import React from 'react';

const Rooms = () => {
  return (
    <div>
      <h1>Rooms</h1>
    </div>
  );
};

export default Rooms;