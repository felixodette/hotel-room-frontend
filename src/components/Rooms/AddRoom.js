import React, { useState } from 'react';

function AddRoom() {
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [image, setImage] = useState('');
  const [size, setSize] = useState('');
  const [view, setView] = useState('');
  const [bedding, setBedding] = useState('');

  const submitRoom = (e) => {
    e.preventDefault();
    const room = {
      name,
      description,
      image,
      size,
      view,
      bedding,
    };

    const requestOptions = {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(room),
    };
    fetch('http://localhost:3000/api/v1/rooms/', requestOptions)
      .then((response) => response.json());

    setName('');
    setDescription('');
    setImage('');
    setSize('');
    setView('');
    setBedding('');
  };
  return (
    <div>
      <h2>Add Room</h2>
      <form
        onSubmit={submitRoom}
      >
        <div>
          <label htmlFor="name">
            Name:
            <input
              id="name"
              type="text"
              placeholder="Name"
              value={name}
              onChange={(e) => setName(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="description">
            Room Description:
            <input
              id="description"
              type="text"
              placeholder="Description"
              value={description}
              onChange={(e) => setDescription(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="image">
            Room Image:
            <input
              id="image"
              type="text"
              placeholder="Image URL"
              value={image}
              onChange={(e) => setImage(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="size">
            Room Size:
            <input
              id="size"
              type="text"
              placeholder="In squaremeter"
              value={size}
              onChange={(e) => setSize(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="view">
            Room View:
            <input
              id="view"
              type="text"
              placeholder="Sea, ocean.."
              value={view}
              onChange={(e) => setView(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="bedding">
            Bedding Option:
            <input
              id="bedding"
              type="text"
              placeholder="Double, triple, twin..."
              value={bedding}
              onChange={(e) => setBedding(e.target.value)}
              required
            />
          </label>

        </div>
        <button type="submit">
          Add
        </button>
      </form>
    </div>
  );

}
  export default AddRoom;

