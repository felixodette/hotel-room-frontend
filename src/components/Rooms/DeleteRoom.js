import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getRooms, deleteRoom } from '../../redux/rooms';
import '../styles/DeleteRoom.css';

const DeleteRoom = () => {
  const dispatch = useDispatch();
  const rooms = useSelector((state) => state.rooms.rooms);
  const [message, setMessage] = useState('');

  const checkResponse = (response, id) => {
    if (response === 204) {
      dispatch(deleteRoom(id));
    } else {
      setMessage(<div className="add-room-error-notification border mt-1 py-1 px-2 bg-dark text-danger rounded p-1">ERROR: Reserved rooms can`t be deleted</div>);
    }
  };
  setInterval(() => { setMessage(''); }, 5000);

  const deleteRoomHandler = (id) => {
    const requestOptions = {
      method: 'DELETE',
      headers: { 'Content-type': 'application/json; charset=UTF-8' },
    };

    fetch(`http://localhost:3000/api/v1/rooms/${id}`, requestOptions)
      .then((response) => checkResponse(response.status, id));
  };

  useEffect(() => {
    dispatch(getRooms());
  }, [dispatch]);
  return (
    <div id="delete-room-container" className="container-fluid d-flex flex-column align-items-center h-100 mb-5">
      <h2 id="add-room-heading" className="text-center  mt-5 fw-bold fs-1 text-white text-uppercase">Delete Room</h2>
      <hr id="add-room-hr" />
      <p className="text-center  fs-6 text-white">
        Would you like to delete your hotel room from our website?
        {' '}
        <br />
        You can do it with one click!
      </p>
      <div>
        {message}
      </div>
      <div id="delete-room-table" className="table w-50 table-hover">
        <table className="table w-100 table-hover">
          <thead>
            <tr>
              <th scope="col" className="text-white">Room Name</th>
              <th scope="col" className="text-center text-white">Action</th>
            </tr>
          </thead>
          <tbody>
            {rooms.map((room) => (
              <tr className="room" key={room.id}>
                <td><p className="text-white mt-3" id="room-name">{room.name}</p></td>
                <td className="text-center">
                  <button
                    id="delete-room-btn"
                    className="mt-2"
                    type="button"
                    onClick={() => deleteRoomHandler(room.id)}
                  >
                    Delete
                  </button>
                </td>
              </tr>
            ))}
          </tbody>

        </table>
      </div>
    </div>
  );
};

export default DeleteRoom;
