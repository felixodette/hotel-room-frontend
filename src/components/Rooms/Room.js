import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';

const Room = () => {
  const [roomDetails, setRoomDetails] = useState('');
  const { id } = useParams();

  useEffect(() => {
    const roomData = {
      method: 'GET',
      headers: { 'Content-Type': 'application/json' },
    };
    fetch(`http://localhost:3001/api/v1/rooms/${id}`, roomData)
      .then((response) => response.json())
      .then((data) => setRoomDetails(data));
  }, []);

  return (
    <div>
      <button type="button">Reserve</button>
      <h1>
        {' '}
        Room name:
        {roomDetails.name}
        {' '}
      </h1>
      <h1>
        {' '}
        Room description:
        {roomDetails.description}
        {' '}
      </h1>
      <h1>
        <img src={roomDetails.image} alt={roomDetails.name} />
      </h1>
      <h1>
        {' '}
        Room size:
        {roomDetails.size}
        {' '}
      </h1>
      <h1>
        {' '}
        Room view:
        {roomDetails.view}
        {' '}
      </h1>
      <h1>
        {' '}
        Room bedding:
        {roomDetails.bedding}
        {' '}
      </h1>
    </div>
  );
};
export default Room;
