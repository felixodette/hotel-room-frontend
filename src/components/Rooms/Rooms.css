.room-card:hover {
    /* change the background shadow coloe */
    box-shadow: 0 0 10px 0 #97BF11;
    /* change the background color */
    background-color: #f1f1f1;
}

.room-description {
    font-size: 1.2rem;
}

.room-image {
    width: 250px;
    height: 250px;
}

.social-icons:hover {
    color: #fff;
    background-color: #000;
}
