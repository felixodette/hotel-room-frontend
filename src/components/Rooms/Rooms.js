import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { GrFacebookOption } from 'react-icons/gr';
import { FaTwitter } from 'react-icons/fa';
import { AiFillLinkedin } from 'react-icons/ai';
import { getRooms } from '../../redux/rooms';

const Rooms = () => {
  const dispatch = useDispatch();
  const rooms = useSelector((state) => state.rooms.rooms);

  useEffect(() => {
    dispatch(getRooms());
  }, [dispatch]);
  return (
    <div>
      <h1 className="text-center mt-5 mb-2 pt-5 fw-bold fs-1">Rooms</h1>
      <h2 className="text-center mt-2 mb-5 fw-bold opacity-50 fs-5">Please select a Room</h2>
      <div className="row row-cols-1 row-cols-md-3 g-4">
        {rooms.map((room) => (
          <a href={`/room/${room.id}`} key={room.id} className="col text-decoration-none text-dark">
            <div className="card h-100 shadow border-0 rounded-3 bg-light text-center p-3 p-md-5 transition" key={room.id}>
              <img src={room.image} alt={room.name} className="card-img-top rounded-3 mb-3 mb-md-4 mx-auto" style={{ width: '300px', height: '200px' }} />
              <div className="card-body">
                <h3 className="card-title fw-bold fs-5 mb-3 mb-md-4">{room.name}</h3>
                <p className="card-text fs-6 opacity-50 mb-3 mb-md-4 text-start text-md-center text-lg-start text-xl-center" style={{ fontSize: '1rem' }}>
                  {' '}
                  {room.description}
                </p>
              </div>
              <div className="card-footer bg-transparent border-0">
                <GrFacebookOption className="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle p-2 mx-2" />
                <FaTwitter className="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle p-2 mx-2" />
                <AiFillLinkedin className="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle p-2" />
              </div>
            </div>
          </a>
        ))}
      </div>
    </div>
  );
};

export default Rooms;
