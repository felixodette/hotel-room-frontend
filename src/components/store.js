import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../redux/reducers/index';

const initialState = {};
const middleware = [thunk];
const store = createStore(
  rootReducer,
  initialState,
  compose(
    applyMiddleware(...middleware),
/* eslint-disable */
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()));
/* eslint-enable */

export default store;
