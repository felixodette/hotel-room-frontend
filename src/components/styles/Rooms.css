#rooms-cotainer {
  min-height: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.room-card:hover {
  box-shadow: 0 0 10px 0 #96bf1165;
}

.room-description {
  font-size: 1rem;
}

#room-image {
  width: 280px;
  height: 200px;
}

.social-icons:hover {
  color: #fff;
  background-color: #000;
}

.carousel-dots {
  bottom: 5%;
}

@media (min-width: 768px) {
  .carousel-dots {
    margin-left: 18%;
  }
}
