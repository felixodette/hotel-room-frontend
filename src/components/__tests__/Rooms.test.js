import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rooms from '../Rooms/Rooms';
import store from '../../redux/configureStore';

describe('Rooms', () => {
  render(
    <Provider store={store}>
      <Rooms />
    </Provider>,
  );

  test('Test the dom for success rendering elements', () => {
    const title = screen.getByText('ROOMS');
    expect(title).toBeInTheDocument();
  });

  test('Test the dom for non existing element on the page', () => {
    const title = screen.queryByText('example');
    expect(title).not.toBeInTheDocument();
  });

  test('It renders the Rooms component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Rooms />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  test('It renders the Rooms component correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rooms />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
