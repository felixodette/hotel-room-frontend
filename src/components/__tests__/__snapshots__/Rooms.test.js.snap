// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms It renders the Rooms component correctly 1`] = `
<div>
  <div
    class="w-100"
  >
    <h1
      class="text-center mt-5 mb-2 pt-5 fw-bold fs-1"
    >
      ROOMS
    </h1>
    <h2
      class="text-center mt-2 mb-5 fw-bold opacity-50 fs-5"
    >
      Please select a Room
    </h2>
    <div
      class="react-multi-carousel-list carousel-container ms-2 me-2 pt-5 pb-5 "
      dir="ltr"
    >
      <ul
        class="react-multi-carousel-track "
        style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
      />
    </div>
    ;
  </div>
</div>
`;

exports[`Rooms It renders the Rooms component correctly 2`] = `
<div
  className="w-100"
>
  <h1
    className="text-center mt-5 mb-2 pt-5 fw-bold fs-1"
  >
    ROOMS
  </h1>
  <h2
    className="text-center mt-2 mb-5 fw-bold opacity-50 fs-5"
  >
    Please select a Room
  </h2>
  <div
    className="react-multi-carousel-list carousel-container ms-2 me-2 pt-5 pb-5 "
    dir="ltr"
  >
    <ul
      className="react-multi-carousel-track "
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>
  ;
</div>
`;
